### PR TITLE
CI: add build & test workflow

### LINKED ISSUE #1: Add GitHub actions for build + tests

### PR DESCRIPTION
**Pull request by p-l-**
Fixes #1

### CODE REVIEW

**Comment by p-l-** [comment]
https://github.com/p-l-/masscanned/runs/4461916303

**Comment by Frky** [comment]
Thank you for the PR. I suggest we also add the built-in unit tests:
```
cargo test
```
What do you think?

**Comment by p-l-** [comment]
> Thank you for the PR. I suggest we also add the built-in unit tests

Good point, done.

https://github.com/p-l-/masscanned/actions/runs/1555951058

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![Build masscanned](https://github.com/p-l-/masscanned/actions/workflows/test.yml/badge.svg?branch=master)](https://github.com/p-l-/masscanned/actions/workflows/test.yml?branch=master)
```
Comment (Frky) [inline README.md:1]: Shouldn't it be https://github.com/ivre/masscanned/...?

File: test/src/all.py [hunk 2]
```diff
@@ -29,10 +29,12 @@
 LOG.setLevel(logging.DEBUG)
 LOG.addHandler(ch)
 
-tests = list()
+tests = []
+errors = []
 
 # decorator to automatically add a function to tests
 def test(f):
+    global errors
```
Comment (Frky) [inline test/src/all.py:37]: Factorize with line 40?

**Review by Frky — COMMENTED** [review]
I added two minor remarks on your commit.

**Comment by p-l-** [comment]
> I added two minor remarks on your commit.

Thanks!

https://github.com/p-l-/masscanned/actions/runs/1556121663

**Comment by p-l-** [comment]
I have cleaned a bit the Python code, also.

https://github.com/p-l-/masscanned/actions/runs/1556264882

**Comment by Frky** [comment]
Thank you for the CI, very useful.

**Pull request merged**